gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/resources/AccountEndpointHandler.java

```diff
@@ -81,6 +99,53 @@ public void getActivity(RoutingContext routingContext) {
                 );
     }
 
+    public void changePassword(RoutingContext routingContext) {
+        final Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
+        final User user = getUserFromContext(routingContext);
+        final DefaultUser principal = convertUserToPrincipal(routingContext, user);
+
+        // update user password value from parameters
+        HttpServerRequest request = routingContext.request();
+        String password = request.getParam(ConstantKeys.PASSWORD_PARAM_KEY);
+
+        Optional<PasswordSettings> passwordSettings = PasswordSettings.getInstance(client, this.domain);
+
+        try {
+            passwordValidator.validate(password, passwordSettings.orElse(null));
+            user.setPassword(password);
+
+            userService.resetPassword(client, user, principal)
```

Comment:
I would prefer refactoring to accountService.resetPassword(user, client, password);